packages/app/webpack.config.js

```diff
@@ -23,13 +29,23 @@ module.exports = {
         ]
     },
     resolve: {
-        extensions: ['.ts', '.tsx', '.mjs', '.js', '.json']
+        extensions: ['.ts', '.tsx', '.mjs', '.js', '.jsx', '.json'],
+        alias:{
+            "react-dom":"@hot-loader/react-dom"
+        }
     },
     plugins: [
+        //new webpack.DefinePlugin({'process.env.NODE_ENV': JSON.stringify(process.env.NODE_ENV)}),
+        new webpack.HotModuleReplacementPlugin(),
+        new webpack.NamedModulesPlugin(),
         new HtmlWebpackPlugin({ title: 'Sample Monorepo App' }),
         new MiniCssExtractPlugin({
             filename: '[name].css',
             chunkFilename: '[name].css'
         })
     ]
 };
+
+//instead of dev/prod configs, i usually do a base, dev, and prod all in 1 file, and do something like config = env === 'prod' ? prod : dev
```

Comment:
I usually use the function-style config: https://webpack.js.org/configuration/configuration-types#exporting-a-function

Unlike what being shown in their docs, you can detect the current mode more clearly via the second parameter:
```js
module.exports = function (_env, {mode = 'development'}) {
    return {
        mode
    };
};
```
This way, if somebody runs `yarn webpack`, it defaults to development mode.
`yarn webpack --mode production` switches to `production`.